guides/05_custom-components/07_pdf-component-example.md

```diff
@@ -0,0 +1,672 @@
+                 every: float | None = None):
+        super().__init__(value, label=label, info=info,
+                         show_label=show_label, container=container,
+                         scale=scale, min_width=min_width,
+                         interactive=interactive, visible=visible,
+                         elem_id=elem_id, elem_classes=elem_classes,
+                         render=render, load_fn=load_fn, every=every)
+        self.height = height
+
+    def preprocess(self, payload: FileData) -> str:
+        return payload.path
+
+    def postprocess(self, value: str | None) -> FileData:
+        if not value:
+            return None
+        return FileData(path=value)
+
+    def example_inputs(self):
+        return "https://gradio-builds.s3.amazonaws.com/assets/pdf-guide/fw9.pdf"
+
+    def as_example(self, input_data: str | None) -> str | None:
+        if input_data is None:
+            return None
+        return processing_utils.move_resource_to_block_cache(input_data, self)
+```
+
+## Step 10: Add a demo and publish!
+
+To test our backend code, let's add a more complex demo that performs Document Question and Answering with huggingface transformers.
+
+In our `demo` directory, create a `requirements.txt` file with the following packages
+
+```
+torch
+transformers
+pdf2image
+pytesseract
+```
+
+
+Tip: Remember to install these yourself and restart the dev server! You may need to install extra non-python dependencies for `pdf2image`. See [here](https://pypi.org/project/pdf2image/). Feel free to write your own demo if you have trouble.
+
+
+```python
+import gradio as gr
+from gradio_pdf import PDF
+from pdf2image import convert_from_path
+from transformers import pipeline
+from pathlib import Path
+
+dir_ = Path(__file__).parent
+
+p = pipeline(
+    "document-question-answering",
+    model="impira/layoutlm-document-qa",
+)
+
+def qa(question: str, doc: str) -> str:
+    img = convert_from_path(doc)[0]
+    output = p(img, question)
+    return sorted(output, key=lambda x: x["score"], reverse=True)[0]['answer']
+
+
+demo = gr.Interface(
+    qa,
+    [gr.Textbox(label="Question"), PDF(label="Document")],
+    gr.Textbox(),
+)
+
+demo.launch()
+```
+
+See our demo in action below!
+
+<video autoplay muted loop>
+  <source src="https://gradio-builds.s3.amazonaws.com/assets/pdf-guide/PDFDemo.mov" type="video/mp4" />
+</video>
+
+Finally lets build our component with `gradio cc build` and publish it with the `gradio cc publish` command!
+This will guide you through the process of uploading your component to [PyPi](https://pypi.org/) and [HuggingFace Spaces](https://huggingface.co/spaces).
+
+
+Tip: You may need to add the following lines to the `Dockerfile` of your HuggingFace Space.
+
+```Dockerfile
+RUN mkdir -p /tmp/cache/
+RUN chmod a+rwx -R /tmp/cache/
+RUN apt-get update && apt-get install -y poppler-utils tesseract-ocr
+
+ENV TRANSFORMERS_CACHE=/tmp/cache/
+```
+
+## Conclusion
```

Comment:
Perhaps we can add a quick section on how to use this component in a Gradio app.

Reply:
Yea good idea!

Reply:
Done

Reply:
Nit: you could use `gr.Interface` with `examples` to showcase the `Example` version of the component